lkmpg.tex

```diff
@@ -228,35 +228,85 @@ \subsection{The Simplest Module}
 	make -C /lib/modules/$(shell uname -r)/build M=$(PWD) clean
 \end{code}
 
+In \verb|Makefile|, \verb|$(CURDIR)| can set to the absolute pathname of the current working directory(after all \verb|-C| options are processed, if any). See more about \verb|CURDIR| in \href{https://www.gnu.org/software/make/manual/make.html}{GNU make manual}.
+
 And finally just run \verb|make| directly.
+
 \begin{codebash}
 make
 \end{codebash}
-If there is no \verb|PWD := $(CURDIR)| statement in Makefile, note that using \verb|sudo make| might not compile. Because the \verb|sudo| will fork the process. According to the Process Model section in \href{https://man7.org/linux/man-pages/man8/sudo.8.html}{sudo manual}, some of the environment variables won't be inherited. Here is the \verb|PWD| variable which will be passed to the kernel Makefile. For example, here is the simple Makefile:
+
+If there is no \verb|PWD := $(CURDIR)| statement in Makefile, note that using \verb|sudo make| might not compile correctly. 
+Because some environment variables can't be inherited, which is specified by the security policy. 
+The default security policy is \verb|sudoers|.
+In the \verb|sudoers| security policy, \verb|env_reset| is enabled by default, which restricts environment variables. 
+Specifically, path variables are not retained from the user environment, they are set to default values (For more information see: \href{https://www.sudo.ws/docs/man/sudoers.man/}{sudoers manual}).
+You can see the environment variable settings by:
+
+\begin{codebash}
```

Comment:
The message was not intended to be a valid bash script. Don't use `codebash` annotation.

Reply:
I see, it should just use `code`?

Reply:
Thanks for your correction. I find something similar, it uses `verbatim`.

> \begin{verbatim}
> $ sudo insmod xxxxxx.ko
> loading out-of-tree module taints kernel.
> module license 'unspecified' taints kernel.
> \end{verbatim}

Reply:
Affirmative. Use `verbatim` for consistency.